server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -345,17 +358,426 @@ public void deleteRegisteredModel(String fullName, boolean force) {
+      throw new BaseException(ErrorCode.INVALID_ARGUMENT, "No updated fields defined.");
+    }
+
+    String fullName = updateModelVersion.getFullName();
+    Long version = updateModelVersion.getVersion();
+    LOGGER.info("Updating Model Version: " + fullName + "/" + version);
+    ModelVersionInfo modelVersionInfo;
+
+    Transaction tx;
+    try (Session session = SESSION_FACTORY.openSession()) {
+      String[] parts = RepositoryUtils.parseFullName(fullName);
+      String catalogName = parts[0];
+      String schemaName = parts[1];
+      String registeredModelName = parts[2];
+      tx = session.beginTransaction();
+      try {
+        // Get the registered model record from the database
+        UUID schemaId = getSchemaId(session, catalogName, schemaName);
+        RegisteredModelInfoDAO existingRegisteredModel =
+            getRegisteredModelDao(session, schemaId, registeredModelName);
+        if (existingRegisteredModel == null) {
+          throw new BaseException(
+              ErrorCode.NOT_FOUND, "Registered model not found: " + registeredModelName);
+        }
+        // Get the model version record from the database
+        ModelVersionInfoDAO origModelVersionInfoDAO =
+            getModelVersionDao(session, existingRegisteredModel.getId(), version);
+        if (origModelVersionInfoDAO == null) {
+          throw new BaseException(
+              ErrorCode.NOT_FOUND, "Model version not found: " + fullName + "/" + version);
+        }
+        origModelVersionInfoDAO.setComment(updateModelVersion.getComment());
+        long updatedTime = System.currentTimeMillis();
+        origModelVersionInfoDAO.setUpdatedAt(new Date(updatedTime));
+        session.persist(origModelVersionInfoDAO);
+        modelVersionInfo = origModelVersionInfoDAO.toModelVersionInfo();
+        modelVersionInfo.setCatalogName(catalogName);
+        modelVersionInfo.setSchemaName(schemaName);
+        modelVersionInfo.setModelName(registeredModelName);
+        tx.commit();
+      } catch (RuntimeException e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    } catch (RuntimeException e) {
+      if (e instanceof BaseException) {
+        throw e;
+      }
+      throw new BaseException(
+          ErrorCode.INTERNAL, "Error updating model version: " + fullName + "/" + version, e);
+    }
+    return modelVersionInfo;
+  }
+
+  public void deleteModelVersion(String fullName, Long version) {
+    LOGGER.info("Deleting model version: " + fullName + "/" + version);
+    String[] parts = fullName.split("\\.");
+    if (parts.length != 3) {
+      throw new BaseException(
+          ErrorCode.INVALID_ARGUMENT, "Invalid registered model name: " + fullName);
+    }
+    String catalogName = parts[0];
+    String schemaName = parts[1];
+    String registeredModelName = parts[2];
+    RegisteredModelInfo rmInfo = getRegisteredModel(fullName);
```

Comment:
Maybe we don't need this check since we would throw in the transaction below.

Reply:
Can we move everything to the transaction. Also since this is the repository class we should deal in DAO's rather than Info objects.